logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/GetNotificationsUseCase.kt

```diff
@@ -61,6 +62,7 @@ internal class GetNotificationsUseCaseImpl internal constructor(
                     .map { list -> list.filter { it.messages.isNotEmpty() } }
             }
             .distinctUntilChanged()
+            .filter { it.isNotEmpty() }
```

Comment:
we had that before, and we removed it due to the user profile update issue! that issue is not valid anymore?

Reply:
The user profile issue was that by flatMapping the `User` -> `Notifications`, when the `User` changed, we would cause the creation of a new flow that would emit the same values as before.

The fact that we filter empty notifications or not doesn't really matter to that problem.

Reply:
thanks for the details, don't know but it would be nice to have such a test to make sure that won't happen.

Reply:
Yeaaah. I want to bring a NotificationManager to inside Kalium actually, make it work on `UserSessionScope`, to make sure there's a app-lifecycle-independent worker doing this kind of things, instead of `ViewModel`, `PersistentWebSocketService`, and push notification receiver fighting over notifications.

The current `WireNotificationManager` in Reloaded is messy and used by too many entities and requires the caller to be very careful.